xcdat/dataset.py

```diff
@@ -316,125 +316,38 @@ def decode_time(dataset: xr.Dataset) -> xr.Dataset:  # noqa: C901
     coord_keys = _get_all_coord_keys(ds, "T")
 
     for key in coord_keys:
-        coord = ds[key].copy()
+        coords = ds[key].copy()
 
-        if not _is_decoded(coord.values):
-            attrs = coord.attrs
-
-            # NOTE: The "calendar" and "units" attributes are stored in the `.attrs`
-            # dict when `decode_times=False`, unlike `decode_times=True` which stores
-            # them in the`.encoding` dict. xcdat handles decoding non-CF time units by
-            # setting xarray to `decode_times=False`. To replicate xarray's decoding
-            # behavior, this function pops the "calendar" and "units" from the `.attrs`
-            # dict and stores them in the `.encoding` dict.
-            calendar = attrs.pop("calendar", None)
-            units_attr = attrs.pop("units", None)
+        if _is_decodable(coords) and not _is_decoded(coords):
+            calendar = coords.attrs.get("calendar", None)
 
             if calendar is None:
+                calendar = "standard"
                 logger.warning(
-                    f"{coord.name} does not have a 'calendar' attribute set, so it could "
-                    "not be decoded. Set the 'calendar' attribute "
-                    f"(`ds.{coord.name}.attrs['calendar]`) and try decoding them again"
+                    f"'{coords.name}' does not have a calendar attribute set. "
+                    "Defaulting to CF 'standard' calendar."
                 )
```

Comment:
I updated this in the latest commit so that the calendar attribute is set directly on the time coordinates rather than a variable (for persistence).

https://github.com/xCDAT/xcdat/blob/135c886ec16ee1fae6995f83e643a1b27e6928c3/xcdat/dataset.py#L325-L330